app/components/inputFields/SelectOptionTextField.tsx

```diff
@@ -0,0 +1,61 @@
+import { MenuItem, TextField } from "@mui/material";
+import React, { FC } from "react";
+import USER_ROLE_OPTIONS from "@/app/constantData/userRoleOption";
+
+interface SelectOptionTextFieldProps {
+  defaultValue?: string;
+  label?: string;
+  helperText?: string;
+}
+
+const SelectOptionTextField: FC<SelectOptionTextFieldProps> = ({
+  defaultValue = "student",
+  label = "Profession",
+  helperText = "Please select your Role",
+}) => {
```

Comment:
Improve Props Handling

Allow USER_ROLE_OPTIONS to be passed as a prop for reusability.
Define explicit types for onChange.

```suggestion
  label = "Profession",
  helperText = "Please select your Role",
  options = USER_ROLE_OPTIONS, // Allows passing custom options
}) => {
```

Reply:
done